File: android/src/oldarch/com/appodeal/rnappodeal/RNAppodealBannerViewManager.java
package com.appodeal.rnappodeal;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

public class RNAppodealBannerViewManager extends SimpleViewManager<RCTAppodealBannerView> {

    private final RNAppodealBannerViewManagerImpl bannerViewManagerImpl;

    public RNAppodealBannerViewManager() {
        this.bannerViewManagerImpl = new RNAppodealBannerViewManagerImpl();
    }

    @NonNull
    @Override
    public String getName() {
        return RNAppodealBannerViewManagerImpl.NAME;
    }

    @NonNull
    @Override
    protected RCTAppodealBannerView createViewInstance(@NonNull ThemedReactContext reactContext) {
        return bannerViewManagerImpl.createViewInstance(reactContext);
    }

    @Override
    public void onDropViewInstance(@NonNull RCTAppodealBannerView view) {
        super.onDropViewInstance(view);
        bannerViewManagerImpl.onDropViewInstance(view);
    }

    @Nullable
    @Override
    public Map<String, Object> getExportedViewConstants() {
        return bannerViewManagerImpl.getExportedViewConstants();
    }

    @ReactProp(name = "adSize")
    public void setAdSize(RCTAppodealBannerView view, @Nullable String value) {
        if (value != null) {
            view.setAdSize(value);
        }
    }

    @ReactProp(name = "placement")
    public void setPlacement(RCTAppodealBannerView view, @Nullable String value) {
        if (value != null) {
            view.setPlacement(value);
        }
    }

    @ReactProp(name = "usesSmartSizing")
    public void setUsesSmartSizing(RCTAppodealBannerView view, boolean value) {
        // No-op for old arch
    }
} 

File: android/src/newarch/com/appodeal/rnappodeal/RNAppodealBannerViewManager.kt
package com.appodeal.rnappodeal

import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewManagerDelegate
import com.facebook.react.viewmanagers.RNAppodealBannerViewManagerDelegate
import com.facebook.react.viewmanagers.RNAppodealBannerViewManagerInterface

@ReactModule(name = RNAppodealBannerViewManagerImpl.NAME)
class RNAppodealBannerViewManager() :
    SimpleViewManager<RCTAppodealBannerView>(),
    RNAppodealBannerViewManagerInterface<RCTAppodealBannerView> {

    private val bannerViewManagerDelegate: ViewManagerDelegate<RCTAppodealBannerView> =
        RNAppodealBannerViewManagerDelegate<RCTAppodealBannerView, RNAppodealBannerViewManager>(this)

    private val bannerViewManagerImpl = RNAppodealBannerViewManagerImpl()

    override fun getDelegate(): ViewManagerDelegate<RCTAppodealBannerView> =
        bannerViewManagerDelegate

    override fun getName(): String = RNAppodealBannerViewManagerImpl.NAME

    override fun createViewInstance(reactContext: ThemedReactContext): RCTAppodealBannerView {
        return bannerViewManagerImpl.createViewInstance(reactContext)
    }

    override fun onDropViewInstance(view: RCTAppodealBannerView) {
        super.onDropViewInstance(view)
        return bannerViewManagerImpl.onDropViewInstance(view)
    }

    override fun getExportedViewConstants(): Map<String?, Any?>? {
        return bannerViewManagerImpl.getExportedViewConstants()
    }

    override fun setAdSize(view: RCTAppodealBannerView, value: String?) {
        value?.let { view.adSize = it }
    }

    override fun setPlacement(view: RCTAppodealBannerView, value: String?) {
        value?.let { view.placement = it }
    }

    override fun setUsesSmartSizing(view: RCTAppodealBannerView, value: Boolean) {
        // No-op
    }
}